pkg/workers/konnectors/konnector.go

```diff
@@ -162,33 +174,57 @@ func Worker(ctx context.Context, m *jobs.Message) error {
 	scanOut := bufio.NewScanner(cmdOut)
 	scanOut.Buffer(nil, 256*1024)
 
-	go doScanOut(jobID, scanOut, domain)
+	var msgChan = make(chan konnectorMsg)
+	var messages []konnectorMsg
+
+	go doScanOut(jobID, scanOut, domain, msgChan)
 	go doScanErr(jobID, scanErr)
+	go func() {
+		hub := realtime.GetHub()
+		for msg := range msgChan {
+			messages = append(messages, msg)
+			hub.Publish(&realtime.Event{
+				Type: realtime.EventCreate,
+				Doc: couchdb.JSONDoc{Type: consts.JobEvents, M: map[string]interface{}{
+					"type":    msg.Type,
+					"message": msg.Message,
+				}},
+				Domain: domain,
+			})
+		}
+	}()
 
 	if err = cmd.Start(); err != nil {
 		return wrapErr(ctx, err)
 	}
 	if err = cmd.Wait(); err != nil {
 		return wrapErr(ctx, err)
 	}
+	for _, msg := range messages {
+		if msg.Type == konnectorMsgTypeError {
+			err = errors.New(msg.Message)
+			return err
+		}
+	}
+	close(msgChan)
 	return nil
 }
 
-func doScanOut(jobID string, scanner *bufio.Scanner, domain string) {
-	hub := realtime.GetHub()
+func doScanOut(jobID string, scanner *bufio.Scanner, domain string, msgs chan konnectorMsg) {
 	for scanner.Scan() {
-		doc := couchdb.JSONDoc{Type: consts.JobEvents}
-		err := json.Unmarshal(scanner.Bytes(), &doc.M)
-		if err != nil {
-			log.Warnf("[konnector] %s: Could not parse Stdout as JSON: %s", jobID, err)
-			log.Warnf("[konnector] %s: Stdout: %s", jobID, scanner.Text())
-			continue
+		linebb := scanner.Bytes()
+		from := bytes.IndexByte(linebb, '{')
+		to := bytes.LastIndexByte(linebb, '}')
+		var msg konnectorMsg
+		log.Infof("[konnector] %s: Stdout: %s", jobID, string(linebb[from:to+1]))
```

Comment:
I don't think you can safely use `linebb[from:to+1]` before checking `from` and `to`